include/crocoddyl/core/utils/exception.hpp

```diff
@@ -26,13 +26,16 @@
     throw crocoddyl::Exception(ss.str(), __FILE__, __PRETTY_FUNCTION__, __LINE__); \
   }
 
+#ifdef NDEBUG
+#define assert_pretty(condition, m) ((void)0)
```

Comment:
Please do not add and merge this to master. Have you been able to see speed-ups with this line added, and if so, how much? The risk of someone developing code in `RelWithDebInfo` or another mode where `NDEBUG` may be added as a flag is too high.

Reply:
Is this a policy in crocoddyl to keep asserts active ALL the time? If so, I didn't know that.

I noticed a high number of cache data misses in the asserts when analyzing the code. Normally the asserts are disabled in release mode, are we not following the same principle here?

Reply:
> I noticed a high number of cache data misses in the asserts when analyzing the code.  
What's the bottom-line impact here concretely?

Often code is in Release or RelWithDebInfo or from the binaries (!) when developing or debugging problem formulations e.g. using Python or dependent code. We had _weeks_ of time lost due to slight bugs or argument errors which did not show or notice. If this has a noticeable effect for one application, it can be activated using a compile-time switch locally but it should not be the default.

Reply:
We  want to keep the assert in release mode as @wxmerkt explained.

Reply:
I'm okay with that, I thought this was an error since `<assert.h>` has a different policy.
I'll revert this commit.
Thanks for the feedback @wxmerkt.

For a problem of 4DoF robot, with 100 nodes and no multithreading, I saw an benefit of ~10us. I'm using Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz,